### PR TITLE
fix: stabilize drive delete E2E terminal-state checks

### PR DESCRIPTION
**Pull request by luozhixiong01**
## Summary

`TestDrive_DeleteAsyncWorkflow` assumed every successful `drive +delete` returns an async `task_id`, so it flaked when Drive transiently reported `drive task failed` for a freshly created docx, and it would also fail on a synchronous delete (no `task_id`). The helper now checks the actual terminal state: it succeeds if the resource is already gone, otherwise it retries the verified transient with a bounded attempt count — while keeping all unrecognized errors fatal.

## Changes

- Rework `deleteAsyncAndVerify` in `tests/cli_e2e/drive/drive_delete_workflow_test.go`: bounded retry loop (3 attempts) covering asynchronous success, synchronous success (empty `task_id` with `deleted=true`), and the verified `drive task failed` transient; add `isTransientDriveDeleteFailure` so only that one error may fall through to terminal-state checking; make `drive +task_result` verification fail fast on non-zero exit
- Add `tests/cli_e2e/drive/drive_delete_workflow_helper_test.go`: fake-CLI contract tests covering error-classification boundaries, exact delete/metadata/task-result call counts per scenario, and subprocess tests proving unexpected delete failures, task-result failures, and retry exhaustion all fail the workflow
- Update `tests/cli_e2e/drive/coverage.md` to describe the terminal-state convergence

## Test Plan

- [x] `make unit-test` passed
- [x] validate passed (build, vet, unit, integration)
- [ ] skipped: sandbox local-eval (test-only change with no new E2E or skill-eval artifacts; the modified live suite runs in this PR's e2e-live job)
- [x] acceptance-reviewer passed (2/2 cases; includes a mutation check that disabling the transient guard turns the subprocess contract test red)
- [x] manual verification: `go test ./tests/cli_e2e/drive/ -run 'TestDeleteAsyncAndVerify|TestIsTransientDriveDeleteFailure|TestDeleteDriveResourceAndVerify' -count=1 -race` — all green against the fake CLI, no real API calls

## Related Issues

N/A

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Drive deletion workflow handling for both synchronous and asynchronous completion, including correct task polling behavior.
  * Added targeted retry handling for a specific transient backend failure until the resource is confirmed deleted.
  * Strengthened verification of deletion outcomes and failure modes.

* **Tests**
  * Expanded end-to-end coverage for deleting docx, empty folders, and non-empty folders across sync/async/transient scenarios.
  * Added more rigorous checks for task-result success, retry exhaustion, and immediate abort behavior on unexpected failures.

* **Documentation**
  * Updated E2E CLI coverage notes for the Drive `delete` command.

### CODE REVIEW

File: tests/cli_e2e/drive/drive_delete_workflow_helper_test.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+		counters := setupFakeWorkflowCounters(t)
+		withFastDeleteWorkflowBackoff(t)
+
+		taskID := deleteAsyncAndVerify(t, context.Background(), "docx_retry", "docx")
+		assert.Equal(t, "task_123", taskID)
+
+		assert.Equal(t, "2", readFakeCounter(t, counters.deletes))
+		assert.Equal(t, "2", readFakeCounter(t, counters.metas), "one terminal-state check after the failure plus the final visibility wait")
+		assert.Equal(t, "1", readFakeCounter(t, counters.taskResults), "async success must verify the task result")
+	})
```
[the remark this answers is not in the document]
Comment (luozhixiong01) [reply tests/cli_e2e/drive/drive_delete_workflow_helper_test.go:65]: ✅ Fixed in fd5bfdac: added subprocess boundary cases for both — a non-zero `+task_result` exit aborts before visibility polling (call counts delete/task-result/meta = 1/1/0), and three transient failures stop at `deleteWorkflowMaxAttempts` (counts 3/3/0).

File: tests/cli_e2e/drive/drive_delete_workflow_test.go [hunk 2]
```diff
@@ -65,30 +66,113 @@ func createDeleteWorkflowDoc(t *testing.T, ctx context.Context, folderToken, tit
...
 	})
 	require.NoError(t, err)
-	taskResult.AssertExitCode(t, 0)
+	require.NotNil(t, taskResult)
+	// Fatal exit-code gate first: the non-fatal assert flavor would cascade
+	// into misleading empty-stdout failures, exactly what this fix removes.
+	require.Equal(t, 0, taskResult.ExitCode, "drive +task_result failed\nstdout:\n%s\nstderr:\n%s", taskResult.Stdout, taskResult.Stderr)
 	taskResult.AssertStdoutStatus(t, true)
 	require.Equal(t, taskID, gjson.Get(taskResult.Stdout, "data.task_id").String(), "stdout:\n%s", taskResult.Stdout)
 	require.False(t, gjson.Get(taskResult.Stdout, "data.failed").Bool(), "stdout:\n%s", taskResult.Stdout)
```
[the remark this answers is not in the document]
Comment (luozhixiong01) [reply tests/cli_e2e/drive/drive_delete_workflow_test.go]: ✅ Fixed in fd5bfdac: `assertDriveDeleteTaskSucceeded` now requires `data.failed` to exist before asserting it is false (the `task_check` scenario always emits the field, so no legitimate absence case).

**Review by liangshuo-1 — APPROVED** [review]
